[BUGFIX] Fix Race Dataset Checksum bug

abarbosa94 opened the pull request:
In #537 I noticed that there was a bug in checksum checking when I have tried to download the race dataset. The reason for this is that the current preprocessing was just considering the `high school` data and it was ignoring the `middle` one. This PR just fixes it :)

Moreover, I have added some descriptions.

lhoestq commented:
I'm not sure this would fix #537 .
However your point about the missing `middle` data is right and we probably want to include these data as well.
Do you think it would we worth having different configurations for this dataset for users who want to only load part of it (`high school` or `middle` or `all`) ?

abarbosa94 commented:
This has fixed #537 at least on my machine hahaha.

Nice point! I think it would totally worth it :) What the best implementation approach would you suggest?

Would it be possible to have `high school`, `middle` and  `all` inside each portion of `train`, `validation` and `test`? Would this make sense?

lhoestq commented:
I think we could have one dataset configuration for `high school`, one for `middle` and one for `all`.
You just need to add

```python

    BUILDER_CONFIGS = [
        HansardsConfig(
            name="house",
            description="""\
          Alignment of debates in the House of the 36th Canadian Parliament: 1,070K sentence pairs.
          """,
        ),
        HansardsConfig(
            name="senate",
            description="""\
          Alignment of debates in the Senate of the 36th Canadian Parliament: 208K sentence pairs.
          """,
        ),
    ]

abarbosa94 commented:
Hi @lhoestq sorry for the delay in addressing your comments. Thanks for your assistance :)

You were correct as well, as I was using the script without the `datasets/race/dataset_infos.json` file, it did not verify the checksum. I already fix it as well :)

I managed to get everything running smoothly by now. Please let me know if you think that I could improve my solution

lhoestq commented on datasets/race/race.py:
```diff
@@ -61,43 +69,36 @@ def _info(self):
-            filepath = os.path.join(filespath, file)
-            with open(filepath, encoding="utf-8") as f:
+        current_path = Path(train_test_or_eval)
+        files_in_dir = [str(f.absolute()) for f in current_path.glob("**/*.txt")]
```
```suggestion
        files_in_dir = [str(f.absolute()) for f in sorted(current_path.glob("**/*.txt"))]
```
We use sorted to make the files order deterministic

lhoestq commented on datasets/race/dataset_infos.json:
```diff
@@ -1 +1 @@
-{"default": {"description": "Race is a large-scale reading comprehension dataset with more than 28,000 passages and nearly 100,000 questions. The\n dataset is collected from English examinations in China, which are designed for middle school and high school students.\nThe dataset can be served as the training and test sets for machine comprehension.\n\n", "citation": "@article{lai2017large,\n     ...
\ No newline at end of file
+{"default": {"description": "Race is a large-scale reading comprehension dataset with more than 28,000 passages and nearly 100,000 questions. The\n dataset is collected from English examinations in China, which are designed for middle school and high school students.\nThe dataset can be served as the training and test sets for machine comprehension.\n\n", "citation": "@article{lai2017large,\n     ...
```
Could you update the dataset_info please ? So that it includes the new configurations

To do so, please delete this file and regenerate it with 
```
datasets-cli test ./race --save_infos --all_configs
```

abarbosa94 replied:
sure! 👍

abarbosa94 replied:
Done in here d6fbac0 @lhoestq 

However, I was wondering the following: I just tested it locally and noticed that if I tried to use the lib as:
`datasets.load_dataset('race')` instead of `datasets.load_dataset('race', 'all')` it still throws the checksum error :( 

I was expecting some kind of warning/error logging that I should pass either `all`; `high` or `middle`. Have I done something wrong?

Could you please advise me? Then I think that I could fix it :)

lhoestq replied:
`datasets.load_dataset('race')` loads the dataset script that is on master if you installed the project using git clone.
It you try `datasets.load_dataset("./datasets/race")` however it should run the local script that you changed

abarbosa94 replied:
cool! Thanks 👍 

Just check it again. Everything seems ok by now.

lhoestq commented on datasets/race/dataset_infos.json:
```diff
@@ -1 +1 @@
-{"default": {"description": "Race is a large-scale reading comprehension dataset with more than 28,000 passages and nearly 100,000 questions. The\n dataset is collected from English examinations in China, which are designed for middle school and high school students.\nThe dataset can be served as the training and test sets for machine comprehension.\n\n", "citation": "@article{lai2017large,\n     ...
\ No newline at end of file
+{"default": {"description": "Race is a large-scale reading comprehension dataset with more than 28,000 passages and nearly 100,000 questions. The\n dataset is collected from English examinations in China, which are designed for middle school and high school students.\nThe dataset can be served as the training and test sets for machine comprehension.\n\n", "citation": "@article{lai2017large,\n     ...
```
Thanks ofr adding the new configurations to the json file :) 

Could you remove the 'default' configuration from this json file ?
To do so you can delete the json file and regenerate it.
Or you can simply remove this field from the json file ^^

abarbosa94 replied:
75d7278 done! :)

I removed the field by hand. I tried to regenerate the file but for some reason, the `default`section was being generated as well ): Do you have any idea why would this be occurring?

lhoestq replied:
Thanks :)

Actually it was still there because it was reusing a cached version of the json file

lhoestq reviewed (approved):
Thank you for updating the checksums and adding the new configurations :)
